ProcessLib/BoundaryCondition/BoundaryCondition.h

```diff
@@ -32,8 +34,23 @@ class BoundaryCondition
 public:
     //! Applies natural BCs (i.e. non-Dirichlet BCs) to the stiffness matrix
     //! \c K and the vector \c b.
-    virtual void apply(const double t, GlobalVector const& x, GlobalMatrix& K,
-                       GlobalVector& b) = 0;
+    virtual void applyNaturalBC(const double /*t*/, GlobalVector const& /*x*/,
```

Comment:
why not NeumannBC here while Dirichlet BC is used below? if it has to be used, Essential BC should be used instead of Dirichlet.